docs/spec_draft.md

```diff
@@ -627,6 +628,47 @@ matrix, then the behavior is undefined.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.clamp
+
+### Semantics
+
+Clamps every element of the `operand` tensor between a minimum and maximum
+value and produces a `result` tensor. More formally, `result[i0, ..., iR-1]` =
+`minimum(maximum(operand[i0, ..., iR-1], min[i0, ..., iR-1]), max[i0, ..., iR-1])`,
+where `minimum` and `maximum` operations correspond to
+[stablehlo.minimum](#stablehlominimum) and
+[stablehlo.maximum](#stablehlomaximum).
+
+### Inputs
+
+| Name      | Type                         |
+|-----------|------------------------------|
+| `min`     | tensor of any supported type |
+| `operand` | tensor of any supported type |
+| `max`     | tensor of any supported type |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) `min`, `operand`, `max`, and `result` have the same type.
```

Comment:
Just like `select`, this op currently supports a restricted form of broadcasting. Let's investigate if it's essential, and if it is, let's support it in the spec, similarly to `select`.

Reply:
#442

Reply:
TF uses scalar max and min (the example is form XLA operation semantics)
```
let operand: s32[3] = {-1, 5, 9};
let min: s32 = 0;
let max: s32 = 6;

Clamp(min, operand, max) = s32[3]{0, 5, 6};
```

Reply:
Jax also supports boradcasting max and min

Reply:
[Jax code pointer](https://github.com/google/jax/blob/0da02dd41c3b4bb5fb685f1a6d421db2a324c2d5/jax/_src/lax/lax.py#L2997)

Reply:
Thank you for looking into this! Then we'll need to update the formalism and the constraints, similarly to `select`, to account for this.

Reply:
Sure, done.